### ISSUE TITLE
Upper case package name 'Boost' causing issues on Windows

### ISSUE THREAD

**Issue by helmesjo** [conan-community/conan-boost#120]
So basically, the package name is "_**boost**/1.xx.0@conan/stable_", but has been packaged with "_**Boost**/1.xx.0@conan/stable_".

This inconsistency causes an issues on Windows:
```
ERROR: Requested 'boost/1.67.0@conan/stable' but found case incompatible 'Boost'
Case insensitive filesystem can't manage this
```

Could we please repackage it correctly?

Thanks!

**Comment by lasote** [comment]
> Could we please repackage it correctly?

What do you want me to fix exactly? The package is lowercase. Due to windows limitations Conan deny to mix same package name with different casing.

**Comment by helmesjo** [comment]
I think I formulated it badly, but basically what I'm seeing on windows after downloading package `boost/1.66.0@conan/stable`, looking in the cache, is instead `Boost/1.66.0@conan/stable`. This causes the error mentioned above when doing as basic things as `conan search boost/1.66.0@conan/stable`, or even as a requirement in a [`conanfile.py`](https://github.com/helmesjo/conan-thrift/blob/7cbbf86e0478cd175dabc785e3e83da62d7d7824/conanfile.py).

So my speculation is that when the package was created for windows, it was specified with uppercase instead of lowercase... Could this be true?

**Comment by lasote** [comment]
This is weird. The Boost package was renamed to lowercase some time ago. Could be possible that you have a remote or old local cache with the old package?

**Comment by helmesjo** [comment]
Nah I've tried removing the package from local cache, download it again and it's still added as **B**oost:

1. Remove package locally:
```
λ  conan remove boost/1.66.0@conan/stable
Are you sure you want to delete from 'Boost/1.66.0@conan/stable' (yes/no): y
λ  conan search boost/*
There are no packages matching the 'boost/*' pattern
```
2. Download again:
```
λ  conan download "boost/1.66.0@conan/stable" -r conan-center
Downloading conanmanifest.txt
[==================================================] 58B/58B
Downloading conanfile.py
[==================================================] 17.2KB/17.2KB
ERROR: Requested 'boost/1.66.0@conan/stable' but found case incompatible 'Boost'
Case insensitive filesystem can't manage this
```
3. Inspect path in cache:
```
λ  conan info Boost/1.66.0@conan/stable --paths
Boost/1.66.0@conan/stable
    ID: 267704336ef55cc31615a75f0c9b0650a9afb8a9
    BuildID: None
    export_folder: C:\Users\fho\.conan\data\Boost\1.66.0\conan\stable\export
    source_folder: C:/.conan\2tl2qmlj\1
    build_folder: C:/.conan\_x9anou5\1
    package_folder: C:/.conan\ojhzw1ye\1
    Remote: None
    URL: https://github.com/lasote/conan-boost
    License: Boost Software License - Version 1.0. http://www.boost.org/LICENSE_1_0.txt
    Updates: Version not checked
    Creation date: 2018-05-29 14:39:27
...
```

**Comment by helmesjo** [comment]
Actually, after getting back to my windows rig and checking some more I noticed that my user cache (`C:\Users\my_user\.conan`) still had the uppercase *Boost* name (even though I did as described above and ran `conan remove` on all my libraries). Somehow the uppercase Boost folder survived and whenever I downloaded again it reused the same name (which, again, was still uppercase even though the package had changed to lowercase).

My solution was to manually delete the *Boost* folder, and everything in the system cache (`C:/.conan`).

I think this started happening when one of my packages relied on `Boost/1.64.0@conan/stable` (which obviously is still is uppercase), and this caused the inconsistencies.
